grpcutils/grpc_authenticator.go

```diff
@@ -0,0 +1,82 @@
+package grpcutils
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"net/http"
+
+	"github.com/grafana/authlib/authn"
+	"github.com/grafana/authlib/types"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/noop"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+type grpcAuthenticator interface {
+	Authenticate(ctx context.Context) (context.Context, error)
+}
+
+type GAuthenticatorFunc func(context.Context) (context.Context, error)
+
+func (fn GAuthenticatorFunc) Authenticate(ctx context.Context) (context.Context, error) {
+	return fn(ctx)
+}
+
+func NewUnsafeAuthenticator() grpcAuthenticator {
```

Comment:
```suggestion
func NewUnsafeAuthenticator(tracer trace.Tracer) grpcAuthenticator {
```
Caller can choose to pas a noop tracer or a real one I guess